sw/device/tests/sram_ctrl_sleep_sram_ret_contents_impl.c

```diff
@@ -188,32 +185,59 @@ bool execute_sram_ctrl_sleep_ret_sram_contents_test(bool scramble) {
              scramble ? "" : "not ");
 
     if (scramble) {
-      dif_sram_ctrl_t ret_sram;
-      addr =
-          mmio_region_from_addr(TOP_EARLGREY_SRAM_CTRL_RET_AON_REGS_BASE_ADDR);
-      CHECK_DIF_OK(dif_sram_ctrl_init(addr, &ret_sram));
+      // NOTE:
+      // The ES version of the ROM does not set the version field of the
+      // retention SRAM when it boots after being scrambled. The ROM_EXT
+      // expects a valid version and refuses to boot otherwise.
+      //
+      // This is fixed in the prod ROM, but we can must work around it in ES
+      // tests by restoring the version after the wipe.
+      //
+      // We write the version pre-scrambling since we want it to be readable
+      // using the default scrambling keys on the next boot.
+      uint32_t version = retention_sram_get()->version;
+
+      LOG_INFO("Wiping ret_sram...");
+      CHECK_STATUS_OK(sram_ctrl_testutils_wipe(&ret_sram));
+
+      retention_sram_get()->version = version;
+
+      LOG_INFO("Scrambling ret_sram...");
+      CHECK_STATUS_OK(sram_ctrl_testutils_scramble(&ret_sram));
+    }
+
+    retention_sram_write();
+    set_up_reset_request();
+  } else if (rstmgr_reset_info & kDifRstmgrResetInfoWatchdog) {
+    LOG_INFO("watchdog reset");
+    // reset due to a reset request, if scramble data is not preserved.
+    retention_sram_check(/* equal = */ !scramble);
+
+    if (scramble) {
+      // NOTE:
+      // As above, we restore the retention SRAM version here, except this time
+      // we do it _after_ scrambling since the next boot will be done using the
+      // new scrambling keys.
+      uint32_t version = retention_sram_get()->version;
+
       LOG_INFO("Wiping ret_sram...");
       CHECK_STATUS_OK(sram_ctrl_testutils_wipe(&ret_sram));
       LOG_INFO("Scrambling ret_sram...");
       CHECK_STATUS_OK(sram_ctrl_testutils_scramble(&ret_sram));
+
+      // Restore the retention SRAM version from before it was scrambled.
+      retention_sram_get()->version = version;
     }
-    test_ret_sram_in_normal_sleep();
```

Comment:
Is this removal intentional?

Reply:
Nope, it was a mistake. Thanks for catching.